### PR TITLE
Kanban card creation followup

### PR DESCRIPTION
**Pull request by ehconitin**
@Bonapara #7002
@FelixMalfait  #6316 
;)

Naming could be improved, do let me know!

https://github.com/user-attachments/assets/b10c9120-644d-4943-bc65-ec0d62f9986f

### CODE REVIEW

**Comment by Bonapara** [comment]
Hi @ehconitin, thanks!

I see there is still an issue with the inline-field height (https://github.com/twentyhq/twenty/issues/7163) on this branch. The relation and tag fields are 28px high instead of 24px. All inline fields should be 24px high.

Also, can you shift the floating field slightly to the right? It should be above the name chip rather than sticking to the left border.

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexPageKanbanAddButton.tsx [hunk 1]
```diff
@@ -30,13 +29,24 @@ const StyledDropDownMenu = styled(DropdownMenu)`
...
 export const RecordIndexPageKanbanAddButton = () => {
   const dropdownId = `record-index-page-add-button-dropdown`;
-  const [isSelectingCompany, setIsSelectingCompany] = useState(false);
-  const [selectedColumnDefinition, setSelectedColumnDefinition] =
-    useState<RecordBoardColumnDefinition>();
 
   const { recordIndexId, objectNamePlural } = useContext(
     RecordIndexRootPropsContext,
   );
+  const { objectNameSingular } = useObjectNameSingularFromPlural({
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexPageKanbanAddButton.tsx:36]: you can probably get it directly from the context above

File: packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts [hunk 2]
```diff
@@ -1,75 +1,124 @@
...
+      if (!columnDefinitionId) {
+        throw new Error('Column ID is required');
+      }
+      return columnDefinitionId;
+    },
+    [columnContext],
+  );
+
+  const addNewOpportunity = useCallback(
+    (set: any, columnDefinitionId: string, position: 'first' | 'last') => {
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts:33]: I agree with Greptile, could you try setting the right type?

File: packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts [hunk 3]
```diff
@@ -1,75 +1,124 @@
 import { RecordBoardContext } from '@/object-record/record-board/contexts/RecordBoardContext';
 import { RecordBoardColumnContext } from '@/object-record/record-board/record-board-column/contexts/RecordBoardColumnContext';
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts:2]: I don't understand why there is so much logic in this file? Couldn't you plug it back into AddNewCard? It feels like you are duplicating code between the two and creating things specific to Opportunity that might not be needed (e.g. do we really need to create `recordBoardNewOpportunityByColumnIdSelector` or could we use the existing
`recordBoardNewRecordByColumnIdSelector` instead? Maybe I'm missing something)

**Review by FelixMalfait — COMMENTED** [review]
Looks great! Can't wait to merge.

File: packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts (hunk 3 above)
Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-board/record-board-column/hooks/useAddNewOpportunity.ts:2]: Yup! I felt the same way too!
Ill fix this!

**Comment by ehconitin** [comment]
@Bonapara could I get a confirmation on should inline fields be 24px all across the app or just kanban cards?

Thanks :)

File: packages/twenty-front/src/modules/object-record/record-board/states/recordBoardNewRecordByColumnIdComponentFamilyState.ts [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 import { createComponentFamilyState } from '@/ui/utilities/state/component-state/utils/createComponentFamilyState';
 
 export type NewCard = {
   id: string;
   columnId: string;
   isCreating: boolean;
   position: 'first' | 'last';
+  isOpportunity: boolean;
+  company: EntityForSelect | null;
 };
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-board/states/recordBoardNewRecordByColumnIdComponentFamilyState.ts:11]: @FelixMalfait How about this? used the same state with additional keys for opportunity creation!

Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/record-board/states/recordBoardNewRecordByColumnIdComponentFamilyState.ts:11]: Oh yes I haven't looked at the code in details again yet but that seems much better :)

Love that you removed more lines that you added 🙌
Screenshot 2024-09-27 at 17 34 19

File: packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx [hunk 5]
```diff
@@ -55,11 +55,9 @@ const StyledInlineCellBaseContainer = styled.div`
   box-sizing: border-box;
   width: 100%;
   display: flex;
-
+  height: 24px;
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx:58]: Agreed! Will do!

[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx:58]: LGTM!

Screenshot from 2024-09-28 14-09-16
Screenshot from 2024-09-28 14-16-48
Screenshot from 2024-09-28 14-17-25

[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx:58]: Great!

**Review by FelixMalfait — APPROVED** [review]
This is great 🙌!

**Pull request merged**